code/modules/asset_cache/asset_list_items.dm

```diff
@@ -640,3 +640,32 @@
 		if (icon != 'icons/misc/language.dmi')
 			var/icon_state = initial(L.icon_state)
 			insert_icon("language-[icon_state]", uni_icon(icon, icon_state))
+
+/// Maps icon names to ref values
+/datum/asset/json/icon_ref_map
+	name = "icon_ref_map"
+	early = TRUE
+
+/datum/asset/json/icon_ref_map/generate()
```

Comment:
(non blocking) this proc is extremely slow btw, it takes around 4 seconds to fully generate

Might be worth offloading walking the file tree to rustg since it's such a simple operation

Reply:
This is part of the Ctrl+C Ctrl+V "port" so i'm not even sure what it does. I not even sure what rustg is so i'm gonna need help with that